ProjectMacros.cmake

```diff
@@ -359,7 +339,7 @@ macro(MAKE_SWIG_TARGET NAME SIMPLENAME KEY_I_FILE I_FILES PARENT_TARGET PARENT_S
   #endif()
   set_target_properties(${swig_target} PROPERTIES LIBRARY_OUTPUT_DIRECTORY "${CMAKE_LIBRARY_OUTPUT_DIRECTORY}/ruby/")
   set_target_properties(${swig_target} PROPERTIES RUNTIME_OUTPUT_DIRECTORY "${CMAKE_RUNTIME_OUTPUT_DIRECTORY}/ruby/")
-  target_link_libraries(${swig_target} ${PARENT_TARGET})
+  target_link_libraries(${swig_target} ${${PARENT_TARGET}_depends})
```

Comment:
This looks fragile. I may have missed it above, but can we use something like
`get_target_properties` to get the _depends list from the PARENT_TARGET (I think there's a change right now it's just relying on us maintaining the same naming conventions in 12 different files)?